core/modules/search/templates/_results_normal.inc.php

```diff
@@ -148,7 +148,7 @@
                         <?php echo (!$issue->hasEstimatedTime()) ? '-' : \thebuggenie\core\entities\Issue::getFormattedTime($issue->getEstimatedTime(true, true)); ?>
                     </td>
                     <td class="sc_spent_time<?php if (!$issue->hasSpentTime()): ?> faded_out<?php endif; ?>"<?php if (!in_array('spent_time', $visible_columns)): ?> style="display: none;"<?php endif; ?>>
-                        <?php echo (!$issue->hasSpentTime()) ? '-' : \thebuggenie\core\entities\Issue::getFormattedTime($issue->getSpentTime(true, true)); ?>
+                        <?php echo (!$issue->hasSpentTime() || !$issue->isSpentTimeVisibleByPermission()) ? '-' : \thebuggenie\core\entities\Issue::getFormattedTime($issue->getSpentTime(true, true)); ?>
```

Comment:
I agree with your logic - as long as the differences between the methods are clear, so it doesn't confuse people too much when looking through the code.